pandas/core/dtypes/missing.py

```diff
@@ -29,23 +29,53 @@
 
 
 def isna(obj):
-    """Detect missing values (NaN in numeric arrays, None/NaN in object arrays)
+    """Detect missing values for an array-like object.
+
+    This function takes an array-like object, for each element, if it is
+    a missing value (`NaN` in numeric arrays, `None`/`NaN` in object arrays)
+    the correisponding element of the output boolean array will be `True`,
+    otherwise `False`.
 
     Parameters
     ----------
-    arr : ndarray or object value
-        Object to check for null-ness
+    obj : array-like or object
+        Object to check for null-ness.
 
     Returns
     -------
-    isna : array-like of bool or bool
+    array-like of bool or bool
```

Comment:
Could you rephrase as `bool or array-like of bool`, so that it goes small to big?